File: server/models/Recording.js
const mongoose = require('mongoose');

const recordingSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  filename: {
    type: String,
    required: true
  },
  originalName: {
    type: String,
    required: true
  },
  filePath: {
    type: String,
    required: true
  },
  duration: {
    type: Number, // Duration in seconds
    default: 0
  },
  transcription: {
    type: String,
    default: ''
  },
  isEmergency: {
    type: Boolean,
    default: false
  },
  emergencyType: {
    type: String,
    enum: ['help', 'medical', 'fire', 'police', 'accident', 'other'],
    default: null
  },
  confidence: {
    type: Number,
    min: 0,
    max: 1,
    default: 0
  },
  keywords: [{
    type: String
  }],
  status: {
    type: String,
    enum: ['processing', 'completed', 'failed'],
    default: 'processing'
  },
  assemblyAiId: {
    type: String,
    default: null
  },
  metadata: {
    fileSize: Number,
    mimeType: String,
    sampleRate: Number
  }
}, {
  timestamps: true
});

// Index for better query performance
recordingSchema.index({ userId: 1, createdAt: -1 });
recordingSchema.index({ isEmergency: 1, createdAt: -1 });

module.exports = mongoose.model('Recording', recordingSchema);

File: flutter_frontend/lib/models/recording.dart
import 'package:json_annotation/json_annotation.dart';

part 'recording.g.dart';

@JsonSerializable()
class Recording {
  @JsonKey(name: '_id')
  final String id;
  final String userId;
  final String filename;
  final String? originalName;
  final String? filePath;
  final String? transcription;
  final bool isEmergency;
  final String? emergencyType;
  final double? confidence;
  final List<String>? keywords;
  final String status;
  final RecordingMetadata? metadata;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Recording({
    required this.id,
    required this.userId,
    required this.filename,
    this.originalName,
    this.filePath,
    this.transcription,
    required this.isEmergency,
    this.emergencyType,
    this.confidence,
    this.keywords,
    required this.status,
    this.metadata,
    required this.createdAt,
    this.updatedAt,
  });

  factory Recording.fromJson(Map<String, dynamic> json) =>
      _$RecordingFromJson(json);
  Map<String, dynamic> toJson() => _$RecordingToJson(this);
}

@JsonSerializable()
class RecordingMetadata {
  final int? fileSize;
  final String? mimeType;
  final int? duration;
  final int? sampleRate;

  RecordingMetadata({
    this.fileSize,
    this.mimeType,
    this.duration,
    this.sampleRate,
  });

  factory RecordingMetadata.fromJson(Map<String, dynamic> json) =>
      _$RecordingMetadataFromJson(json);
  Map<String, dynamic> toJson() => _$RecordingMetadataToJson(this);
}

@JsonSerializable()
class EmergencyAnalysis {
  final List<String> matchedKeywords;
  final String emergencyType;
  final double confidenceScore;
  final String riskLevel;

  EmergencyAnalysis({
    required this.matchedKeywords,
    required this.emergencyType,
    required this.confidenceScore,
    required this.riskLevel,
  });

  factory EmergencyAnalysis.fromJson(Map<String, dynamic> json) =>
      _$EmergencyAnalysisFromJson(json);
  Map<String, dynamic> toJson() => _$EmergencyAnalysisToJson(this);
}